src/main/java/uk/gov/digital/ho/hocs/domain/service/ProcessingService.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.digital.ho.hocs.domain.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.application.LogEvent;
+import uk.gov.digital.ho.hocs.domain.exceptions.ApplicationExceptions;
+import uk.gov.digital.ho.hocs.domain.queue.common.MessageHandler;
+
+import javax.validation.constraints.NotNull;
+import java.time.LocalDateTime;
+
+@Service
+@Slf4j
+public class ProcessingService {
+
+    private final MessageLogService messageLogService;
+
+    private final MessageHandler messageHandler;
+
+    public ProcessingService(MessageLogService messageLogService, MessageHandler messageHandler) {
+        this.messageLogService = messageLogService;
+        this.messageHandler = messageHandler;
+    }
+
+    public void processMessages(int maxMessages, LocalDateTime from, @NotNull LocalDateTime to) {
+        var messageCount = messageLogService.getCountOfPendingMessagesBetweenDates(from, to);
+        if (messageCount > maxMessages) {
+            throw new ApplicationExceptions.TooManyMessagesException(
+                    String.format("Too many messages to process. Maximum allowed: %s. Current: %s", maxMessages, messageCount),
+                    LogEvent.TOO_MANY_MESSAGES);
+        }
+
+        // retrieve messages between from and to dates
+        var messages = messageLogService.getPendingMessagesBetweenDates(from, to);
+        messages.forEach(message -> {
+            try {
+                messageHandler.handleMessage(message.getMessage(), message.getMessage());
+                messageLogService.completeMessageLogEntry(message.getMessageId());
+            } catch (Exception e) {
+                log.error("Error processing message: {}. For reason: {}.", message.getMessageId(), e.getMessage());
```

Comment:
We're going to want a flag here for after processing to throw an exception if a message fails - unless we want that to be another cronjob to check statuses. Feels better to be part of this though.